app/src/androidTest/java/ch/epfl/balelecbud/authentication/MockAuthenticator.java

```diff
@@ -32,169 +25,75 @@
         }
     };
 
+    private static int uid = 0;
+
+    private User currentUser;
+
     private MockAuthenticator() {
         loggedIn = false;
     }
 
     @Override
-    public void signIn(final String email, final String password, Callback callback) {
+    public CompletableFuture<User> signIn(final String email, final String password) {
         if (users.containsKey(email) && users.get(email).equals(password)) {
             setLoggedIn(true);
-            callback.onSuccess();
+            return MockDatabaseWrapper.getInstance().getDocument("users", "0", User.class).thenCompose(new Function<User, CompletionStage<User>>() {
```

Comment:
why not use thenApply and hide the CompletionStage ?

Reply:
You're right, it makes more sense to use thenApply here.